### PR TITLE
#2070: variant: replace SafeUnion with std::variant

### LINKED ISSUE #2070: Replace AlignedCharUnion with `std::variant`
**What Needs to be Done?**
- [ ]  Finish https://github.com/DARMA-tasking/magistrate/issues/274
- [ ]  Replace AlignedCharUnion in https://github.com/DARMA-tasking/vt/blob/develop/src/vt/utils/adt/union.h with `std::variant`

### PR DESCRIPTION
**Pull request by lifflander**
Fixes #2070

### CODE REVIEW

File: src/vt/runtime/runtime_diagnostics.cc [hunk 1]
```diff
@@ -100,22 +100,25 @@ struct FormatHelper {
...
   FormatHelper fn(unit, align, base);
-  return eval.switchOn(fn);
+  std::string out = "";
+  std::visit([&](auto&& t){
+    using T = std::decay_t<decltype(t)>;
+    if constexpr (std::is_same_v<T, double>) {
+      out = fn.apply<double>(eval);
+    } else if constexpr (std::is_same_v<T, int64_t>) {
+      out = fn.apply<int64_t>(eval);
+    } else if constexpr (std::is_same_v<T, uint64_t>) {
```
Comment (thearusable) [inline src/vt/runtime/runtime_diagnostics.cc:117]: Should we keep this `else if` statement as we don't have `uint64_t` in the `UnionValueType`?

File: src/vt/runtime/component/diagnostic_value.cc [hunk 2]
```diff
@@ -109,14 +109,7 @@ void reduceHelper(
...
 
 DIAGNOSTIC_VALUE_INSTANCE(int64_t)
-DIAGNOSTIC_VALUE_INSTANCE(int32_t)
-DIAGNOSTIC_VALUE_INSTANCE(int16_t)
-DIAGNOSTIC_VALUE_INSTANCE(int8_t)
-DIAGNOSTIC_VALUE_INSTANCE(uint64_t)
-DIAGNOSTIC_VALUE_INSTANCE(uint32_t)
-DIAGNOSTIC_VALUE_INSTANCE(uint16_t)
-DIAGNOSTIC_VALUE_INSTANCE(uint8_t)
+// DIAGNOSTIC_VALUE_INSTANCE(uint64_t)
```
Comment (thearusable) [inline src/vt/runtime/component/diagnostic_value.cc:112]: I think we can remove this comment.

**Review by thearusable — APPROVED** [review]

**Review by nlslatt — COMMENTED** [review]
I have exactly the same questions/comments as Arek.

**Review by nlslatt — APPROVED** [review]
Looks good

**Pull request merged**